src/components/List/ApplicationsList.tsx

```diff
@@ -79,11 +81,27 @@ interface ApplicationRowProps {
 }
 
 const ApplicationRow: React.FC<ApplicationRowProps> = ({ columns, application }) => {
-  return (
+  const [isDeleted, setIsDeleted] = useState<boolean>(false)
+  const [deleteApplication, { data, loading, error }] = useDeleteApplicationMutation({
+    variables: { id: application.id || 0 },
+  })
+  const props = {
+    application,
+    loading,
+    deleteApplication,
+  }
+
+  useEffect(() => {
+    if (data) setIsDeleted(true)
+  }, [data])
+
+  if (error) return <Message error header={strings.ERROR_APPLICATION_DELETE} list={[error]} />
+
+  return isDeleted ? null : (
```

Comment:
Is there a better way to do this? From what I understand, this just set the local component to "deleted" and just hides it from UI even though it's still actually in the list. What we really want to do is trigger a re-fetch in the list, which means you won't need any state in here.

One way to do it, maybe: 
Use "refetch" in "ListWrapper": `{data, loading, error: applicationsError, refetch } = useGetApplicationListQuery(...`

And pass this down as "refetchList" or something, then in the ApplicationRow component, use the delete mutation's "ON_COMPLETED" property, and call "refetchList" on completion (so no need for useEffect either).

(One side effect of the way you've done it, is if the list is longer than one page, then when you delete one, the list won't re-paginate, which is confusing cos it looks like you've got less applications that there are -- in fact, you could delete all the applications on a page and the user would think they've deleted everything because the existing ones haven't rippled round to the current page like they should have).

Reply:
Doing this change now. Thanks : )